examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/search/blocs/puppy_list_bloc.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:async';
+import 'package:equatable/equatable.dart';
+import 'package:favorites_advanced_base/core.dart';
+import 'package:favorites_advanced_base/repositories.dart';
+
+import 'package:bloc/bloc.dart';
+import 'package:favorites_advanced_base/models.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:meta/meta.dart';
+import 'package:rxdart/rxdart.dart';
+
+part 'puppy_list_event.dart';
+
+part 'puppy_list_state.dart';
+
+// part 'puppy_list_bloc_models.dart';
+
+class PuppyListBloc extends Bloc<PuppyListEvent, PuppyListState> {
+  PuppyListBloc(this.repository)
+      : super(const PuppyListState(
+          searchedPuppies: [],
+          status: PuppyListStatus.initial,
+        ));
+
+  PuppiesRepository repository;
+  var allPuppies = <Puppy>[];
+  late var emptyPuppies = <Puppy>[];
+
+  // var lastFetched = <Puppy>[];
+
+  @override
+  Stream<Transition<PuppyListEvent, PuppyListState>> transformEvents(
+    Stream<PuppyListEvent> events,
+    TransitionFunction<PuppyListEvent, PuppyListState> transitionFn,
+  ) {
+    print('TRANSFORMEVENTS');
+
+    return super.transformEvents(
+        Rx.merge([
+          events.doOnData((event) {
+            print('-- Events: ${event.props}');
+          }),
+          events
+              .whereType<PuppyFetchExtraDetailsEvent>()
+              .mapEventToList()
+              .distinct()
+              .doOnData((event) {
+            print(
+                '-- PuppyListFetchExtraDetailsEvent: ${event.filteredPuppies}');
+          }),
+        ]).distinct().doOnData((event) {
+          print('-- MERGED stream : ${event.props}');
+        }),
+        transitionFn);
+  }
+
+  @override
+  Stream<PuppyListState> mapEventToState(
+    PuppyListEvent event,
+  ) async* {
+    print('mapEventToState : ${event.toString()}');
+    if (event is LoadPuppyListEvent) {
+      yield await _mapPuppiesFetchedToState(state);
+    } else if (event is ReloadPuppiesEvent) {
+      yield await _mapPuppiesReloadFetchToState(state);
+      await Future.delayed(const Duration(milliseconds: 2000));
+    } else if (event is PuppyListFetchExtraDetailsEvent) {
+      if (event.filteredPuppies.isEmpty) {
+        return;
+      }
+      yield await _mapPuppyListFetchExtraDetailsEventToState(
+          event.filteredPuppies, state);
+    }
+  }
+
+  Future<PuppyListState> _mapPuppyListFetchExtraDetailsEventToState(
+      List<Puppy> filteredPuppies, PuppyListState state) async {
+    final puppiesWithDetails = await repository.fetchFullEntities(
+        filteredPuppies.map((element) => element.id).toList());
+
+    allPuppies = allPuppies.mergeWith(puppiesWithDetails);
+
+    return PuppyListState(
```

Comment:
@StanislavPrime, wouldn't be better to use state.copyWith(...) instead of creating brand new PuppyListState ?

Reply:
I changed the returning of PuppyListState to state.copyWith().